### PR TITLE
Adding getCookie() and DeleteCookie() methods with appropriate tests

### PR DESCRIPTION
**Pull request by matthewtsirulnikov**

### CODE REVIEW

**Review by AlexeyCL — CHANGES_REQUESTED** [review]

File: src/commons/BrowserUtils.ts [hunk 1]
```diff
@@ -673,6 +673,27 @@ export namespace BrowserUtils {
...
   }
 
+  /**
+   * Get cookie
+   * You can query a specific cookie by providing the cookie name or
+   * retrieve all.
+   */
+  export function getCookie(names?: string[] | string): WebDriver.Cookie[] {
+    const cookie: WebDriver.Cookie[] = tryBlock(() => browser.getCookies(names), 'Failed to get cookie');
+    Reporter.debug(`Get cookie: '${cookie}'`);
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:683]: Done

File: src/commons/BrowserUtils.ts [hunk 2]
```diff
@@ -673,6 +673,27 @@ export namespace BrowserUtils {
     }
   }
 
+  /**
+   * Get cookie
+   * You can query a specific cookie by providing the cookie name or
+   * retrieve all.
+   */
+  export function getCookie(names?: string[] | string): WebDriver.Cookie[] {
+    const cookie: WebDriver.Cookie[] = tryBlock(() => browser.getCookies(names), 'Failed to get cookie');
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:682]: Done

File: src/commons/BrowserUtils.ts [hunk 3]
```diff
@@ -673,6 +673,27 @@ export namespace BrowserUtils {
...
+    return cookie;
+  }
+
+  /**
+   * Delete cookie
+   * By providing a cookie name it just removes the single cookie or more when multiple names are passed.
+   */
+  export function deleteCookie(names?: string[] | string): void {
+    const cookie: WebDriver.Cookie[] = tryBlock(() => browser.deleteCookies(names), 'Failed to get cookie');
+    Reporter.debug(`Get cookie: '${cookie}'`);
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:694]: Done

[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:694]: Done

File: src/test/specs/DeleteCookieSpec.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { assert } from 'chai';
+import { BrowserUtils, TestUtils } from '../..';
+import { describeCommon } from '../TestHelper';
+
+/**
+ * wdio-allure-ts deleteCookie tests
+ */
+let cookie: WebDriver.Cookie;
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/test/specs/DeleteCookieSpec.ts:9]: Done

File: src/test/specs/DeleteCookieSpec.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    browser.setCookies(cookie);
+  });
+  it('expect empty cookie', () => {
+    BrowserUtils.deleteCookie();
+    assert.equal(browser.getCookies().length, 0, 'Incorrect number of retrieved cookies');
+  });
+
+  it('delete all cookie', () => {
+    BrowserUtils.deleteCookie();
+    assert.equal(browser.getCookies()[0], emptyCookie, 'Cookie has not been removed');
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/test/specs/DeleteCookieSpec.ts:24]: Done

File: src/test/specs/DeleteCookieSpec.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ * wdio-allure-ts deleteCookie tests
+ */
+let cookie: WebDriver.Cookie;
+// @ts-ignore
+const emptyCookie: WebDriver.Cookie;
+
+describeCommon('DeleteCookieSpec', () => {
+  beforeEach(() => {
+    cookie = { name: TestUtils.randomString(), value: TestUtils.randomString() };
+    browser.setCookies(cookie);
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/test/specs/DeleteCookieSpec.ts:15]: Done

File: src/commons/BrowserUtils.ts [hunk 7]
```diff
@@ -673,6 +673,27 @@ export namespace BrowserUtils {
     }
   }
 
+  /**
+   * Get cookie
+   * You can query a specific cookie by providing the cookie name or
+   * retrieve all.
+   */
+  export function getCookie(names?: string[] | string): WebDriver.Cookie[] {
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:681]: Done

File: src/commons/BrowserUtils.ts [hunk 8]
```diff
@@ -673,6 +673,27 @@ export namespace BrowserUtils {
...
+    Reporter.debug(`Get cookie: '${cookie}'`);
+
+    return cookie;
+  }
+
+  /**
+   * Delete cookie
+   * By providing a cookie name it just removes the single cookie or more when multiple names are passed.
+   */
+  export function deleteCookie(names?: string[] | string): void {
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:692]: Done

File: src/test/specs/GetCookieSpec.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * wdio-allure-ts getCookie tests
+ */
+let cookie: WebDriver.Cookie;
+
+describeCommon('GetCookieSpec', () => {
+  beforeEach(() => {
+    cookie = { name: TestUtils.randomString(), value: TestUtils.randomString() };
+    browser.setCookies(cookie);
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/test/specs/GetCookieSpec.ts:13]: Done

File: src/test/specs/GetCookieSpec.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+  it('get cookie by name', () => {
+    assert.equal(BrowserUtils.getCookie()[0].name, cookie.name, 'Incorrect cookie name');
+  });
+
+  it('get cookie by value', () => {
+    assert.equal(BrowserUtils.getCookie()[0].value, cookie.value, 'Incorrect cookie value');
+  });
+  afterEach(() => {
+    browser.deleteCookies();
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/test/specs/GetCookieSpec.ts:27]: Done

File: src/commons/BrowserUtils.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts]: also changed the names of the spec files from single to plural

File: src/commons/BrowserUtils.ts [hunk 11]
```diff
@@ -673,6 +673,29 @@ export namespace BrowserUtils {
...
 
+  /**
+   * Get cookie
+   * You can query a specific cookie by providing the cookie name or
+   * retrieve all.
+   */
+  export function getCookies(names?: string[] | string): WebDriver.Cookie[] {
+    Reporter.debug('Get cookie:');
+    const cookie: WebDriver.Cookie[] = tryBlock(() => browser.getCookies(names), 'Failed to get cookie');
+    Reporter.debug(`${JSON.stringify(cookie)}`);
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:684]: ``${}`` its redundant here, you can use without it

File: src/commons/BrowserUtils.ts [hunk 12]
```diff
@@ -673,6 +673,29 @@ export namespace BrowserUtils {
...
+  }
+
+  /**
+   * Delete cookie
+   * By providing a cookie name it just removes the single cookie or more when multiple names are passed.
+   */
+  export function deleteCookies(names?: string[] | string): void {
+    Reporter.debug('Get cookie:');
+    const cookie: WebDriver.Cookie[] = tryBlock(() => browser.deleteCookies(names), 'Failed to get cookie');
+    Reporter.debug(`${JSON.stringify(cookie)}`);
```
Comment (AlexeyCL) [inline src/commons/BrowserUtils.ts:696]: '`${}`' its redundant here, you can use without it

File: src/test/specs/DeleteCookiesSpec.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { assert } from 'chai';
+import { BrowserUtils, TestUtils } from '../..';
+import { describeCommon } from '../TestHelper';
+
+/**
+ * wdio-allure-ts deleteCookie tests
+ */
+let cookie: WebDriver.Cookie;
+const emptyCookie: WebDriver.Cookie = null;
```
Comment (AlexeyCL) [inline src/test/specs/DeleteCookiesSpec.ts:9]: it should be undefiend

File: src/commons/BrowserUtils.ts [hunk 14]
```diff
@@ -673,6 +673,29 @@ export namespace BrowserUtils {
     }
   }
 
+  /**
+   * Get cookie
+   * You can query a specific cookie by providing the cookie name or
+   * retrieve all.
+   */
+  export function getCookies(names?: string[] | string): WebDriver.Cookie[] {
+    Reporter.debug('Get cookie:');
```
[the remark this answers is not in the document]
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:682]: Done

File: src/commons/BrowserUtils.ts (hunk 11 above)
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:684]: Done

File: src/commons/BrowserUtils.ts (hunk 12 above)
Comment (matthewtsirulnikov) [reply src/commons/BrowserUtils.ts:696]: Done

File: src/test/specs/DeleteCookiesSpec.ts (hunk 13 above)
Comment (matthewtsirulnikov) [reply src/test/specs/DeleteCookiesSpec.ts]: Done

**Review by AlexeyCL — APPROVED** [review]

**Pull request merged**